homeassistant/components/shelly/utils.py

```diff
@@ -315,6 +315,14 @@ def get_model_name(info: dict[str, Any]) -> str:
     return cast(str, MODEL_NAMES.get(info["type"], info["type"]))
 
 
+def get_deviceinfo_model(device: BlockDevice | RpcDevice) -> str | None:
+    """Return the device model for deviceinfo."""
+    if isinstance(device, RpcDevice) and device.xmod_info != {}:
+        return cast(str, device.xmod_info.get("n"))
```

Comment:
I think we should only use `n` value if it exists in `xmod_info`.
```suggestion
    if isinstance(device, RpcDevice) and (model := device.xmod_info.get("n")):
        return cast(str, model)
```